sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CognitiveComplexityWalker.cs

```diff
@@ -51,6 +53,25 @@ public void EnsureVisitEndedCorrectly()
             }
         }
 
+        public void Walk(SyntaxNode node)
+        {
+            try
+            {
+                Visit(node);
+            }
+            catch (InsufficientExecutionStackException)
+            {
+                // TODO: trace this exception
+
+                // Roslyn walker overflows the stack when the depth of the call is around 2050.
+                // See ticket #727.
+
+                // Reset nesting level, so the problem with the walker is not reported.
+                nestingLevel = 0;
+                return;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant jump. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3626)